analyzer/src/main/java/org/jboss/gm/analyzer/alignment/io/SettingsFileIO.java

```diff
@@ -100,4 +88,20 @@ private static String extractProjectNameFromScmUrl(File rootDir) throws IOExcept
             throw new IOException("Unable to read .git/config file found, failed to determine the root project name", e);
         }
     }
+
+    // Package private for testing
+    static File findGitDir(File current) {
+        File prospective = new File(current, GIT_CONFIG);
+        File parent = current.getParentFile();
+
+        logger.info("Searching for Git config in {}", prospective);
+        if (prospective.exists()) {
```

Comment:
Better to test `isDirectory()` instead of `exists` which could match a non-directory as well?

Reply:
Git config is a file though?

Reply:
I guess I could have used isFile ....

Reply:
Yes, `isFile()` for `".git" + File.separator + "config"` may be more accurate, and when I said `isDirectory()` I was thinking of finding `.git/`.

Reply:
I'll change it to `isFile`.